classes/submissionFile/Repository.php

```diff
@@ -270,6 +270,7 @@ public function add(SubmissionFile $submissionFile): int
             [
                 'assocType' => PKPApplication::ASSOC_TYPE_SUBMISSION_FILE,
                 'assocId' => $submissionFile->getId(),
+                'userId' => $submissionFile->getUploaderUserId(),
```

Comment:
This should not be overridden here. The `userId` is already defined in "https://github.com/pkp/pkp-lib/blob/main/classes/submissionFile/Repository.php#L836" `getSubmissionFileLogData` function.

This is true for the other occurrences of this addition through this PR

Reply:
For cli mode <which is recommended for import sometimes> iserId stay empty. It calls above mentioned fatal error and import fails.
If overridden (thou should not be) gui mode import tool works samely, but cli mode import will be successful.

Reply:
@Igor-In I understand that this change may add a `userId` in the `add` function, thus resolving your the DB error in this specific case, but changing the functionality of this core function may cause your installation to behave unexpectantly.

The problem in this case is in the calling code (the import/export plugin) - that's why a broader discussion on user management in those cases may be in order.

Reply:
Thanks, @Igor-In! As @defstat suggests, we've got a bit of infrastructural work needed in order to clean up code that assumes it'll be invoked from a user request when it could sometimes be CLI. Meanwhile, I suggest editing the `getSubmissionFileLogData` function so that it doesn't just look to the request for the current user's identification.

Command line tools can allow for a user to be specified there (the `--user_name` option). When that's specified, the user is stored in the `Registry` [under the `user` key](https://github.com/pkp/pkp-lib/blob/fa3c7bf2ead5a202435927a66324735968a549ac/classes/cliTool/CommandLineTool.php#L148). For cases when `Request::getUser` returns `null` in `getSubmissionFileLogData`, this could be used as a fallback.

Reply:
> Thanks, @Igor-In! As @defstat suggests, we've got a bit of infrastructural work needed in order to clean up code that assumes it'll be invoked from a user request when it could sometimes be CLI. Meanwhile, I suggest editing the `getSubmissionFileLogData` function so that it doesn't just look to the request for the current user's identification.
> 
> Command line tools can allow for a user to be specified there (the `--user_name` option). When that's specified, the user is stored in the `Registry` [under the `user` key](https://github.com/pkp/pkp-lib/blob/fa3c7bf2ead5a202435927a66324735968a549ac/classes/cliTool/CommandLineTool.php#L148). For cases when `Request::getUser` returns `null` in `getSubmissionFileLogData`, this could be used as a fallback.

@Igor-In taking into account the note that @asmecher made, please try to execute the CLI command using the `-u` switch - try

```
php tools/importExport.php NativeImportExportPlugin import bio_6v3+.xml bio -u oleg
```

instead of 

```
php tools/importExport.php NativeImportExportPlugin import bio_6v3+.xml bio oleg
```

Reply:
Yes, you are right, that switch is obligatory for user in cli mode, sorry had not founded.
Thank You!

Reply:
@Igor-In not a problem, the documentation on Import/Export may not have been up-to-date with this change, so it's not your fault. Please feel free to contact us for any related problem that you may encounter, and keep your involvement with the code base with possible suggestions for fixes with your contributions.